src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -998,22 +998,19 @@ parselMx' pstep initial extract (Stream step state) = do
             Yield x s -> do
                 pRes <- pstep pst x
                 case pRes of
-                    -- PR.Yield 0 pst1 -> go SPEC s [] pst1
-                    PR.Yield n pst1 -> do
-                        assert (n <= length (x:buf)) (return ())
-                        go SPEC s (Prelude.take n (x:buf)) pst1
-                    PR.YieldB n pst1 -> do
+                    PR.Partial 0 pst1 -> go SPEC s [] pst1
```

Comment:
Is there a reason, you've added this case now, but commented it in the previous one?

Reply:
Maybe, it was an error as you did not comment the one for `Skip`/`Continue`

Reply:
Ok, maybe it was not, You specifically ignored `Yield 0` for some reason.

Reply:
The earlier the general case for Yield was different it was optimized to efficient code in 0 case, now the general case is backtracking which would be inefficient for the 0 case.